CMakeLists.txt

```diff
@@ -220,14 +231,11 @@ endif()
 set(CMAKE_POSITION_INDEPENDENT_CODE TRUE)
 
 #-------------------------------------------------------------------------------
-# IREE compilation toolchain configuration
+# IREE compilation flags
 #-------------------------------------------------------------------------------
 
-# Enable using lld as the linker for C/C++ targets. This affects IREE and all
-# dependency projects.
-option(IREE_ENABLE_LLD "Use lld when linking" OFF)
-
-string(JOIN " " CMAKE_CXX_FLAGS ${IREE_DEFAULT_COPTS})
+string(JOIN " " IREE_DEFAULT_COPTS_STR ${IREE_DEFAULT_COPTS})
```

Comment:
what is this? we don't want to do string manipulation like this as it breaks flags with spaces in them - COPTS is a list (; separated) and should only be manipulated with list()

Reply:
Tweaked from existing code. :) But is `CMAKE_CXX_FLAGS` a list? The cases what I've seen thus far they are all manipulated as a string.. (Which is ugly for sure.) I remembered that treating it with `list(APPEND` seeing errors. Let me do a double check.

Reply:
Treating it as list generates compilation command like `/usr/bin/clang++ ... -fdiagnostics-color -g;-fsanitize=address` and errors out..

Reply:
We should be treating lists as lists... fighting with CMake like this leads to cross-platform issues and pain :/

Reply:
Yeah, agreed. But I think the problem is that `CMAKE_CXX_FLAGS` is treated as a single string instead of a list by CMake when given to the compiler: https://stackoverflow.com/questions/16433391/cmake-list-append-for-compiler-flags-yields-bogus-results. I converted where I can in iree_copts.cmake, but here still we need to concatenate to make a single string. Unless I miss something or another approach to handle this..